qiskit/circuit/tools/pi_check.py

```diff
@@ -45,14 +49,22 @@ def pi_check(inpt, eps=1e-6, output='text', ndigits=5):
         try:
             return pi_check(float(inpt), eps=eps, output=output, ndigits=ndigits)
         except (ValueError, TypeError):
-            return str(inpt)
+            param_str = str(inpt)
+            nums = re.findall(r'\d*\.\d+|\d+', param_str)
```

Comment:
I like to avoid re whenever is possible. Would it be possible here to use `inpt._symbol_expr.expr_free_symbols` to extract the values instead?

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/4807/commits/68cd7376866bbf25859bbe0e113921d25ab85a04.